app/src/main/java/com/epmedu/animeal/tabs/TextInputField.kt

```diff
@@ -0,0 +1,292 @@
+            )
+        }
+    }
+}
+
+class FormattedTransformation(format: String) : VisualTransformation {
+    val mask: String = format
+    val cleanDigitLength = mask.replace("\\D".toRegex(), "").length
+    private val map: MutableMap<Int, Char> = mutableMapOf()
+
+    init {
+        for (a in mask.indices) {
+            if (mask[a] != '1') {
+                map[a - map.size - 1] = mask[a]
+            }
+        }
+        map[cleanDigitLength] = ' '
+    }
+
+    override fun filter(text: AnnotatedString): TransformedText {
+        val trimmed =
+            if (text.text.length >= cleanDigitLength) text.text.substring(0 until cleanDigitLength) else text.text
+        val annotatedString = AnnotatedString.Builder().run {
+            for (i in trimmed.indices) {
+                append(trimmed[i])
+                if (map.containsKey(i)) {
+                    append(map[i].toString())
+                }
+            }
+            mask.takeLast(mask.length - length)
+            toAnnotatedString()
+        }
+        return TransformedText(
+            text = annotatedString,
+            offsetMapping = object : OffsetMapping {
+                override fun originalToTransformed(offset: Int): Int {
+                    var lastIndex = 0
+                    while (true) {
+                        when {
+                            lastIndex == map.keys.size -> {
+                                return mask.length
+                            }
+                            offset <= map.keys.toList()[lastIndex] -> {
+                                return offset + lastIndex
+                            }
+                            else -> {
+                                lastIndex += 1
+                            }
+                        }
+                    }
+                }
+
+                override fun transformedToOriginal(offset: Int): Int {
+                    var lastIndex = 0
+                    while (true) {
+                        when {
+                            lastIndex == map.keys.size -> {
+                                return mask.length
+                            }
+                            offset <= map.keys.toList()[lastIndex] -> {
+                                return offset - lastIndex
+                            }
+                            else -> {
+                                lastIndex += 1
+                            }
+                        }
+                    }
+                }
+            }
+        )
+    }
+}
+
+@Preview(
+    showBackground = true,
+    name = "TextInputField",
+    showSystemUi = true,
+    uiMode = UI_MODE_NIGHT_NO,
+)
+@Composable
+fun TextInputPreview() {
+    AnimealTheme {
+        Surface(color = MaterialTheme.colors.background, modifier = Modifier.fillMaxSize()) {
+            var name by remember { mutableStateOf("") }
+
+            TextInputField(
+                modifier = Modifier
+                    .fillMaxWidth()
```

Comment:
I guess that fillMaxWidth should be set by default, shouldn't it?

Reply:
It doesnot worked so that I did not changed

Reply:
You can add .fillMaxWidth() to modifier for TextField and remove setting horizontalAlignment to Column.
Also you don't need to pass modifier to all children of column, only for column itself.

Reply:
Done, If I have done incorrectly, I will fix